run_tests.py

```diff
@@ -0,0 +1,177 @@
+    MEDIUM = 'medium'  # All but the slowest tests.
+    FAST = 'fast'  # Only fast-running tests (TBD target runtime).
+
+    def __str__(self):
+        return self.value
+
+
+def main():
+    args = get_args()
+
+    # Read arguments
+    blender_execs = get_blender_binaries()
+    if not len(blender_execs):
+        print("ERROR : No Blender exe paths found!")
+        return
+
+    # Compile the addon
+    res = subprocess.check_output(COMPILE_CMD)
+    print(res.decode("utf-8"))
+
+    reset_test_file()
+
+    # Loop over all binaries and run tests.
+    t0 = time.time()
+    for ind, binary in enumerate(blender_execs):
+        run_all = args.all_execs is True
+        run_specific = args.version is not None
+        if ind > 0 and not (run_all or run_specific):
+            continue  # Only run the first test unless specified
+        if not os.path.exists(binary):
+            print(f"Bledner EXE not found: {binary}")
+            continue
+        cmd = [binary,
+               "--background",
+               "--factory-startup",
+               "-y",
+               "--python",
+               TEST_RUNNER,
+               "--"]  # TODO: Option to specify specific test or test class
+        if args.test_specific is not None:
+            cmd.extend(["-t", args.test_specific])
+        if args.version is not None:
+            cmd.extend(["-v", args.version])
+        output = subprocess.check_output(cmd)
+        print(output.decode("utf-8"))
+
+    t1 = time.time()
+
+    output_results()
+    round_s = round(t1 - t0)
+    print(f"tests took {round_s}s to run")
+
+
+def get_args():
+    """Sets up and returns argument parser for module functions."""
+    parser = argparse.ArgumentParser(description="Run MCprep tests.")
+    # Would use action=argparse.BooleanOptionalAction,
+    # but that's python 3.9+ only; we need to support 3.7.0+ (Blender 2.80)
+    parser.add_argument(
+        "-a", "--all_execs", action="store_true",
+        help="Run the test suite once per each executable in blender_execs.txt"
+    )
+    parser.set_defaults(all_execs=False)
+    parser.add_argument(
+        "-t", "--test_specific",
+        help="Run only a specific test function or class matching this name")
+    parser.add_argument(
+        "-v", "--version",
+        help="Specify the blender version(s) to test, in #.# or #.##,#.#")
+    parser.add_argument(
+        "-s", "--speed",
+        type=TestSpeed,
+        choices=list(TestSpeed),
+        help="Which speed of tests to run")
+    return parser.parse_args()
+
+
+def get_blender_binaries() -> List:
+    """Extracts a list of strings meant to be blender binary paths."""
+    if not os.path.exists(DCC_EXES):
+        print(f"ERROR : The {DCC_EXES} file is missing! Please create it!")
+        return []
+
+    with open(DCC_EXES, "r") as f:
+        blender_execs = [ln.rstrip() for ln in f.readlines()]
+
+    return blender_execs
```

Comment:
I think we need to consider the case where the `blender` binary is on developers' `PATH`, as is the case with Blender on Linux systems

Reply:
Lots of room for improvements, though we still want to ensure we encourage developers to have many versions of blender to test their changes against. Given there's no standard way to create aliases to reference different blender versions in paths, I feel the txt file solution here is suitable, even if not elegant.